hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_BlockingTest.java

```diff
@@ -0,0 +1,519 @@
+        warmUpPartitions(factory.getAllHazelcastInstances());
+
+        NodeEngineImpl nodeEngine = getNodeEngineImpl(local);
+
+        String key = generateKeyOwnedBy(remote);
+        int partitionId = nodeEngine.getPartitionService().getPartitionId(key);
+
+        // first we execute an operation that stall the partition.
+        InternalOperationService opService = nodeEngine.getOperationService();
+        opService.invokeOnPartition(null, new SlowOperation(5 * callTimeout), partitionId);
+
+        // then we execute a lock operation that won't be executed because the partition is blocked.
+        LockOperation op = new LockOperation(new InternalLockNamespace(key), nodeEngine.toData(key), 1, -1, -1);
+        InternalCompletableFuture<Object> future = opService.createInvocationBuilder(null, op, partitionId)
+                .setCallTimeout(callTimeout)
+                .invoke();
+
+        // then we register our callback
+        final ExecutionCallback callback = mock(ExecutionCallback.class);
+        future.andThen(callback);
+
+        // and we eventually expect to fail with an OperationTimeoutException
+        assertFailsEventuallyWithOperationTimeoutException(callback);
+    }
+
+    @Test
+    public void sync_whenOperationTimeout() {
+        int callTimeout = 5000;
+        Config config = new Config().setProperty(OPERATION_CALL_TIMEOUT_MILLIS, "" + callTimeout);
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance local = factory.newHazelcastInstance(config);
+        HazelcastInstance remote = factory.newHazelcastInstance(config);
+        warmUpPartitions(factory.getAllHazelcastInstances());
+
+        NodeEngineImpl nodeEngine = getNodeEngineImpl(local);
+
+        String key = generateKeyOwnedBy(remote);
+        int partitionId = nodeEngine.getPartitionService().getPartitionId(key);
+
+        // first we execute an operation that stall the partition.
+        InternalOperationService opService = nodeEngine.getOperationService();
+        opService.invokeOnPartition(null, new SlowOperation(5 * callTimeout), partitionId);
+
+        // then we execute a lock operation that won't be executed because the partition is blocked.
+        LockOperation op = new LockOperation(new InternalLockNamespace(key), nodeEngine.toData(key), 1, -1, -1);
+        InternalCompletableFuture<Object> future = opService.createInvocationBuilder(null, op, partitionId)
+                .setCallTimeout(callTimeout)
+                .invoke();
+
+        // then we register our callback
+        final ExecutionCallback callback = mock(ExecutionCallback.class);
+        future.andThen(callback);
+
+        // and we eventually expect to fail with an OperationTimeoutException
+        assertFailsEventuallyWithOperationTimeoutException(callback);
+    }
+
+    @Test
+    public void async_whenOperationTimeout() {
+//        Config config = new Config();
```

Comment:
Is this test not compiling? If it does I would prefer `@Ignore`, so the test shows up in the statistics and isn't forgotten.